cranelift/codegen/src/isa/x64/inst/emit_tests.rs

```diff
@@ -2560,6 +2560,57 @@ fn test_x64_emit() {
         "movslq  -7(%r11), %rdx",
     ));
 
+    // Mov_Imm_M.
+    insns.push((
+        Inst::MovImmM {
+            size: OperandSize::Size64,
+            simm64: 0u64,
+            dst: Amode::imm_reg(8u32, rsp).into(),
+        },
+        "48C744240800000000",
+        "movq    $0, 8(%rsp)",
+    ));
+
+    insns.push((
+        Inst::MovImmM {
+            size: OperandSize::Size32,
+            simm64: -126i64 as u64,
+            dst: Amode::imm_reg(99u32, r8).into(),
+        },
+        "41C7406382FFFFFF",
+        "movl    $-126, 99(%r8)",
+    ));
+
+    insns.push((
+        Inst::MovImmM {
+            size: OperandSize::Size64,
+            simm64: -126i64 as u64,
+            dst: Amode::imm_reg(99u32, rdi).into(),
+        },
+        "48C7476382FFFFFF",
+        "movq    $-126, 99(%rdi)",
+    ));
+
+    insns.push((
+        Inst::MovImmM {
+            size: OperandSize::Size64,
+            simm64: 1234567u64,
+            dst: Amode::imm_reg(99u32, r9).into(),
+        },
+        "49C7416387D61200",
+        "movq    $1234567, 99(%r9)",
+    ));
+
+    insns.push((
+        Inst::MovImmM {
+            size: OperandSize::Size64,
+            simm64: i32::MIN as u64,
+            dst: Amode::imm_reg(99u32, rbp).into(),
+        },
+        "48C7456300000080",
+        "movq    $-2147483648, 99(%rbp)",
+    ));
+
```

Comment:
Can we add `Size16` and `Size8` cases here too?

Reply:
Done, thanks!